File: Leetcode/455. Assign Cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        int c=0;
        int j=0,i=0;
        while(i<s.size() && j<g.size()){
            if(g[j]<=s[i]){
                c++;
                j++;
            }
            i++;
        }
        return c;
    }
};

File: Leetcode/455. Assign Cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        l,r=0,0
        m,n=len(g),len(s)
        if m==0 or n==0:
            return 0
        while r<n and l<m:
            if g[l]<=s[r]:
                l+=1
            r+=1
        return l
        
